core/src/main/scala/org/http4s/MediaType.scala

```diff
@@ -104,43 +114,50 @@ object MediaRange {
       def main = MediaRangeFull
     }.parse
 
-  private[http4s] trait MediaRangeParser extends Rfc2616BasicRules { self: PbParser =>
+  private[http4s] def renderExtensions(sb: Writer, mr: MediaRange): Unit =
+    mr.extensions.foreach { case (k, v) => sb << ';' << ' ' << k << '=' <<# v }
+
+  private[http4s] def extensionsToString(mr: MediaRange): String = {
+    val sw = new StringWriter()
+    renderExtensions(sw, mr)
+    sw.result
+  }
+
+  private[http4s] trait MediaRangeParser extends MediaParser { self: PbParser =>
     def MediaRangeFull: Rule1[MediaRange] = rule {
       MediaRangeDef ~ optional(oneOrMore(MediaTypeExtension)) ~> {
         (mr: MediaRange, ext: Option[Seq[(String, String)]]) =>
           ext.fold(mr)(ex => mr.withExtensions(ex.toMap))
       }
     }
 
-    def MediaRangeDef: Rule1[MediaRange] = rule {
-      (("*/*" ~ push("*") ~ push("*")) |
-        (Token ~ "/" ~ (("*" ~ push("*")) | Token)) |
-        ("*" ~ push("*") ~ push("*"))) ~> (getMediaRange(_, _))
-    }
-
-    def MediaTypeExtension: Rule1[(String, String)] = rule {
-      ";" ~ OptWS ~ Token ~ optional("=" ~ (Token | QuotedString)) ~> {
-        (s: String, s2: Option[String]) =>
-          (s, s2.getOrElse(""))
-      }
-    }
+    def MediaRangeDef: Rule1[MediaRange] = MediaRangeRule[MediaRange](getMediaRange)
 
     private def getMediaRange(mainType: String, subType: String): MediaRange =
       if (subType === "*")
         MediaRange.standard.getOrElse(mainType.toLowerCase, new MediaRange(mainType))
-      else MediaType.getOrElseCreate((mainType.toLowerCase, subType.toLowerCase))
+      else
+        MediaType.all.getOrElse(
+          (mainType.toLowerCase, subType.toLowerCase),
+          new MediaType(mainType.toLowerCase, subType.toLowerCase))
   }
 
   implicit val http4sInstancesForMediaRange
     : Show[MediaRange] with HttpCodec[MediaRange] with Order[MediaRange] =
     new Show[MediaRange] with HttpCodec[MediaRange] with Order[MediaRange] {
-      override def show(s: MediaRange): String = s.toString
+      override def show(s: MediaRange): String =
+        s"${s.mainType}/*${MediaRange.extensionsToString(s)}"
```

Comment:
I've made `show` match what`s produced by `render` rather than match `toString`. this made it easier to remove `Renderable` from the hierarchy

Reply:
Interesting.  This is something we should be consistent on throughout the model: do we want `show` to match the wire representation and be part of the `HttpCodec` laws?